src/coreclr/tools/Common/Compiler/ObjectWriter/UnixObjectWriter.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Buffers.Binary;
+using ILCompiler.DependencyAnalysis;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.ObjectWriter
+{
+    /// <summary>
+    /// Base implementation for ELF and Mach-O object file format writers. Implements
+    /// the common code for DWARF debugging and exception handling information.
+    /// </summary>
+    internal abstract partial class UnixObjectWriter : ObjectWriter
+    {
+        private sealed record UnixSectionDefinition(string SymbolName, Stream SectionStream);
+        private readonly List<UnixSectionDefinition> _sections = new();
```

Comment:
Nit: this my personal style, but I'm generally not a fan of splitting fields across multiple partial definitions. In almost all cases it's not important, but occassionally the field ordering matters. Also, it's nice to be able to look at one place and generally guess at the size of the type.

That said, just my personal preference, no need to follow it.